design/cli-install-changes.md

```diff
@@ -0,0 +1,286 @@
+        -l, --selector string                             only show items matching this label selector
+        --show-labels                                     show labels in the last column
+
+```
+
+3) `velero snapshot-location`
+Commands/flags for snapshot locations. 
+
+```
+     set 
+        --default mapStringString                         sets the list of unique volume providers and default volume snapshot location (provider1:location-01,provider2:location-02,...) (NEW, -- was `server --efault-volume-snapshot-locations)  
+
+      create
+        --dry-run                                         generate resources, but don't send them to the cluster. Use with -o. Optional. (NEW)
+        --default                                         Sets these new locations to be the new default snapshot locations. Default is false. (NEW) 
+        --secret-file string                              file containing credentials for volume provider. If not specified, set --no-secret must be used for confirmation. Optional. (MOVED FROM install)
+        --no-secret                                       flag indicating if a secret should be created. Must be used as confirmation if create --secret-file is not provided. Optional. (MOVED FROM install)
+        --config  mapStringString                         configuration to use for creating a volume snapshot location. Format is key1=value1,key2=value2 (was also in `velero install --`snapshot-location-config`). Required.
+        --provider string                                 provider name for volume storage. Required.
+        -h, --help                                        help for create
+        --label-columns stringArray                       a comma-separated list of labels to be displayed as columns
+        --labels mapStringString                          labels to apply to the volume snapshot location
+        -o, --output string                               Output display format. For create commands, display the object but do not send it to the server. Valid formats are 'table', 'json', and 'yaml'. 'table' is not valid for the install command.
+        --provider string                                 name of the volume snapshot provider (e.g. aws, azure, gcp)
+        --show-labels                                     show labels in the last column
+
+
+      get                                                 Display snapshot locations
+        --default                                         list of unique volume providers and default volume snapshot location (provider1:location-01,provider2:location-02,...) (NEW -- was `server --default-volume-snapshot-locations`))
+```
+
+4) `velero plugin`
+Configuration for plugins.
+
+```
+      add 
+        --images stringArray                    add plugin container images to install into the Velero Deployment
```

Comment:
Currently, the image name is specified not as a flag, but as an argument the command (e.g. `velero plugin add velero/velero-plugin-for-aws:v1.0.0`).  I'd say this is a standard in our CLI (and others as well), where the "object" of the command is specified not as a flag, but as an argument to the command. It's similar to `velero backup get my-backup`, or `kubectl get pod my-pod`, as opposed to something like `velero backup get --name my-backup` or `kubectl get pod --name my-pod`

I prefer that pattern for consistency and clarity -- so the "object" of a command is an argument, and flags are strictly for modifiers. WDYT?

Reply:
I would agree with Steve's comments here - the image could just be a positional argument, not a flag.

There's one exception that I can think of in Velero, where `velero restore create` can take a schedule or a backup name - thus `--from-backup` or `--from-schedule` are required.

Reply:
Yes! If I ever deviate from the standard it's pure oversight, glad to have it pointed out.

In this case, I moved the `--images` flag from the `velero install` command to here and didn't give it much more thinking. Now that I'm thinking about it: should we have the ability to add more than one plugin simultaneously? If not, I'll scratch this and follow the original formatting!

Reply:
As discussed on the call, it would be nice to support multiple plugins simultaneously, and we could do this similarly to other commands, e.g. `velero plugin add my/image-1 my-image-2 ...`